From patchwork Wed Sep 19 17:19:12 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peng Hao <peng.hao2@zte.com.cn>
X-Patchwork-Id: 10605539
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 5D69D913
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 19 Sep 2018 09:06:12 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 4D25D2A831
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 19 Sep 2018 09:06:12 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 41A672ACD6; Wed, 19 Sep 2018 09:06:12 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 55C362A831
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 19 Sep 2018 09:06:11 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1728334AbeISOnG (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 19 Sep 2018 10:43:06 -0400
Received: from mxhk.zte.com.cn ([63.217.80.70]:35908 "EHLO mxhk.zte.com.cn"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1727056AbeISOnG (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 19 Sep 2018 10:43:06 -0400
Received: from mse01.zte.com.cn (unknown [10.30.3.20])
        by Forcepoint Email with ESMTPS id 487BD475111D3737BC5A;
        Wed, 19 Sep 2018 17:06:08 +0800 (CST)
Received: from notes_smtp.zte.com.cn ([10.30.1.239])
        by mse01.zte.com.cn with ESMTP id w8J960gM081516;
        Wed, 19 Sep 2018 17:06:00 +0800 (GMT-8)
        (envelope-from peng.hao2@zte.com.cn)
Received: from localhost.localdomain.localdomain ([10.74.120.59])
          by szsmtp06.zte.com.cn (Lotus Domino Release 8.5.3FP6)
          with ESMTP id 2018091917061041-126976 ;
          Wed, 19 Sep 2018 17:06:10 +0800
From: Peng Hao <peng.hao2@zte.com.cn>
To: pbonzini@redhat.com, rkrcmar@redhat.com
Cc: kvm@vger.kernel.org, Peng Hao <peng.hao2@zte.com.cn>
Subject: [PATCH v2]  kvm-unit-test: fix script output abnormal
Date: Thu, 20 Sep 2018 01:19:12 +0800
Message-Id: <1537377552-37142-1-git-send-email-peng.hao2@zte.com.cn>
X-Mailer: git-send-email 1.8.3.1
X-MIMETrack: Itemize by SMTP Server on SZSMTP06/server/zte_ltd(Release
 8.5.3FP6|November
 21, 2013) at 2018-09-19 17:06:10,
        Serialize by Router on notes_smtp/zte_ltd(Release 9.0.1FP7|August  17,
 2016) at
 2018-09-19 17:05:52,
        Serialize complete at 2018-09-19 17:05:52
X-MAIL: mse01.zte.com.cn w8J960gM081516
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

./run_tests.sh
)ASS apic-split (50 tests;
)ASS ioapic-split (19 tests;
)ASS apic (50 tests;
)ASS ioapic (19 tests;
)ASS smptest (1 tests;
)ASS smptest3 (1 tests;

The problem come from commit 0f4f2acb2c880607518cb740a6aa06acafd05855
(x86: io: force carriage return on \n put to serial port). It maybe output
well in real hardware, but it let outputs to file with '\r'.

Changes v1 --> v2:
	changes in scripts/runtime.bash can affect other arch.

Signed-off-by: Peng Hao <peng.hao2@zte.com.cn>
---
 lib/x86/io.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/x86/io.c b/lib/x86/io.c
index edac5bc..3be55d7 100644
--- a/lib/x86/io.c
+++ b/lib/x86/io.c
@@ -6,6 +6,7 @@
 #ifndef USE_SERIAL
 #define USE_SERIAL
 #endif
+#undef USE_SERIAL_HARDWARE_OUTPUT
 
 static struct spinlock lock;
 static int serial_iobase = 0x3f8;
@@ -24,9 +25,11 @@ static void serial_outb(char ch)
 
 static void serial_put(char ch)
 {
+#ifdef USE_SERIAL_HARDWARE_OUTPUT
         /* Force carriage return to be performed on \n */
         if (ch == '\n')
                 serial_outb('\r');
+#endif
         serial_outb(ch);
 }
 
